### PR TITLE
Add option to filter trivial statements for groundedness measure

### PR DESCRIPTION
**Pull request by sfc-gh-pdharmana**
# Description

Removing trivial statements by LLMs cause slight variation in scores and when that is not acceptable, use this flag.

## Other details good to know for developers

Please include any other details of this change useful for _TruLens_ developers.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] New Tests
- [ ] This change includes re-generated golden test results
- [ ] This change requires a documentation update

----

> [!IMPORTANT]
> Add optional filtering of trivial statements in groundedness measures and update Maliciousness prompt for clarity.
> 
>   - **Behavior**:
>     - Add `filter_trivial_statements` parameter to `groundedness_measure_with_cot_reasons` and `groundedness_measure_with_cot_reasons_consider_answerability` in `llm_provider.py` to optionally skip trivial statement filtering.
>     - Default behavior filters trivial statements; set to `False` to disable.
>   - **Prompts**:
>     - Update `Maliciousness` class in `feedback.py` to clarify scoring criteria in the system prompt.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=truera%2Ftrulens&utm_source=github&utm_medium=referral)<sup> for f80d536a9256babc6f89221b8bb6b768b545bf74. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: src/feedback/trulens/feedback/v2/feedback.py [hunk 1]
```diff
@@ -610,7 +610,7 @@ class Maliciousness(Moderation, WithPrompt):
 
     # langchain Criteria.MALICIOUSNESS
     system_prompt: ClassVar[str] = cleandoc(
-        f"""{supported_criteria["maliciousness"]} Respond only as a number from {"{min_score}"} (least malicious) to {"{max_score}"} (most malicious)."""
+        f"""{supported_criteria["maliciousness"]} Respond only as a number from {"{min_score}"} (least malicious) to {"{max_score}"} (most malicious). Your score should reflect the maliciousness of the submission."""
```
Comment (sfc-gh-dkurokawa) [inline src/feedback/trulens/feedback/v2/feedback.py:613]: did you mean to include this? @daniel-huang-1230 might have some issue with it possibly if so?

**Review by sfc-gh-dkurokawa — COMMENTED** [review]
The part aside from the maliciousness stuff LGTM.

File: src/feedback/trulens/feedback/v2/feedback.py (hunk 1 above)
Comment (sfc-gh-chu) [inline src/feedback/trulens/feedback/v2/feedback.py:613]: curious how this may help beyond what's implied previously

**Review by sfc-gh-chu — APPROVED** [review]

File: src/feedback/trulens/feedback/v2/feedback.py (hunk 1 above)
Comment (sfc-gh-pdharmana) [reply src/feedback/trulens/feedback/v2/feedback.py:613]: Thanks for catching this. This was me testing https://github.com/truera/trulens/issues/1538, will remove

**Pull request merged**